packages/web-components/src/components/pricing-table/pricing-table.ts

```diff
@@ -28,6 +37,266 @@ class DDSPricingTable extends StableSelectorMixin(DDSStructuredList) {
+          threshold: 0,
+        }
+      );
+
+      this._intersectionObserverStart.observe(_startSentinelNode);
+      this._intersectionObserverEnd.observe(_endSentinelNode);
+    }
+  }
+
+  /**
+   * Safely disconnects and removes the intersection observer.
+   */
+  private _cleanIntersectionObservers() {
+    if (this._intersectionObserverStart instanceof IntersectionObserver) {
+      this._intersectionObserverStart.disconnect();
+    }
+    this._intersectionObserverStart = null;
+
+    if (this._intersectionObserverEnd instanceof IntersectionObserver) {
+      this._intersectionObserverEnd.disconnect();
+    }
+    this._intersectionObserverEnd = null;
+  }
+
+  /**
+   * Returns true if a cell's contents are valid to animate when transitioning * between sticky states and false otherwise.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private _validateCell(cell: HTMLElement): boolean {
+    return (
+      cell instanceof DDSPricingTableHeaderCell &&
+      cell.getAttribute('type') !== PRICING_TABLE_HEADER_CELL_TYPES.SIMPLE &&
+      cell.shadowRoot !== null
+    );
+  }
+
+  /**
+   * Animates specific cell elements when transitioning between sticky states.
+   */
+  private _animateCellElements(cell: HTMLElement) {
+    const { isSticky } = this;
+    const selectors = [`.${prefix}--pricing-table-header-cell-tag-wrapper`, `.${prefix}--pricing-table-cell-inner`];
+
+    if (this._validateCell(cell)) {
+      selectors.forEach(selector => {
+        const element = (cell.shadowRoot as ShadowRoot).querySelector(selector);
+        if (element instanceof HTMLElement) {
+          if (isSticky) {
+            slideHidden(element);
+          } else {
+            slideUnhidden(element);
+          }
+        }
+      });
+    }
+  }
+
+  /**
+   * Animates header cells when transitioning between sticky states.
+   */
+  private _animateCells() {
+    const { headerCells, isSticky } = this;
+    const { cellStickyClass } = this.constructor as typeof DDSPricingTable;
+    headerCells?.forEach(cell => {
+      if (isSticky) {
+        cell.classList.add(cellStickyClass);
+      } else {
+        cell.classList.remove(cellStickyClass);
+      }
+      this._animateCellElements(cell);
+    });
+  }
+
+  /**
+   * Animates the header row when transitioning between sticky states.
+   */
+  private _animateHeaderRow() {
+    const { headerRow, isSticky } = this;
+    const { rowStickyClass } = this.constructor as typeof DDSPricingTable;
+    if (headerRow) {
+      if (isSticky) {
+        headerRow.classList.add(rowStickyClass);
+      } else {
+        headerRow.classList.remove(rowStickyClass);
+      }
+    }
+  }
+
+  /**
+   * Set or unset sticky state of pricing table header row.
+   */
+  private _setSticky(sticky: boolean = !this.isSticky) {
+    this.isSticky = sticky;
+    if (sticky) {
+      // Add buffer to prevent animation jitters.
+      this._startSentinelNode.style.top = '-10px';
```

Comment:
It's all looking good! Maybe we could have a constant for the `-10px` instead?

Reply:
Thanks for the review, @IgnacioBecerra! I set this value in a `_startSentinelBuffer` class property.

Let me know if you want to take a look at the rest of this together. There's a lot to make sense of, and I'd be happy to walk you through it all.